include/torch-mlir/Dialect/Torch/IR/GeneratedTorchOps.td

```diff
@@ -5313,9 +5313,10 @@ def Torch_AtenOnesLikeOp : Torch_Op<"aten.ones_like", [
 }
 
 def Torch_AtenEmptyMemoryFormatOp : Torch_Op<"aten.empty.memory_format", [
+    NoSideEffect,
```

Comment:
Hi, this file is generated and should not be edited manually. Can you please revert this patch so that it does not block other progress, since other people will be running https://github.com/llvm/torch-mlir/blob/main/build_tools/update_torch_ods.sh to update this file and lose your changes and break these tests?

You will need to update this script to make it smart enough to emit these traits:
https://github.com/llvm/torch-mlir/blob/9d6ee48661cefa677950bcd262fe2824a4fa9d17/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py#L435

In this case, I don't think that `aten.empty.memory_format` is NoSideEffect because it can raise an error if the parameters are wrong. Can you find a way to resubmit this PR without needing this?

Reply:
I updated this attribute just because mhlo is value sematic, but it seems `aten.empty.memory_format ` against it. The decomposed pass lowers some patterns such as `aten.empty.memory_format` + `aten.fill`, in mhlo sematic, we can lower the fill op to be a constant tensor, instead of creating an empty tensor and fill it. To eliminate the `aten.empty.memory_format` after Canonicalizer pass, I added the NoSideEffect attribute.

But I agree with you, `aten.empty.memory_format` may raise an error, I will revert this PR and find another way to lower the `native_dropout` op.

Reply:
Could we change the decomposition to use aten::new_full ?

Reply:
It would also be good for us to add a dialect-level canonicalization pattern that erases ReadOnly ops if they don't have uses. That would cover this and many other useful cases.